PostHog/PostHogFeatureFlags.swift

```diff
@@ -127,25 +127,44 @@ class PostHogFeatureFlags {
                 #endif
 
                 self.featureFlagsLock.withLock {
-                    if errorsWhileComputingFlags {
-                        let cachedFeatureFlags = self.getCachedFeatureFlags() ?? [:]
-                        let cachedFeatureFlagsPayloads = self.getCachedFeatureFlagPayload() ?? [:]
+                    let cachedFeatureFlags = self.getCachedFeatureFlags() ?? [:]
+                    let cachedFeatureFlagsPayloads = self.getCachedFeatureFlagPayload() ?? [:]
 
-                        let newFeatureFlags = cachedFeatureFlags.merging(featureFlags) { _, new in new }
-                        let newFeatureFlagsPayloads = cachedFeatureFlagsPayloads.merging(featureFlagPayloads) { _, new in new }
+                    let newFeatureFlags: [String: Any]
+                    let newFeatureFlagPayloads: [String: Any]
 
+                    if errorsWhileComputingFlags {
                         // if not all flags were computed, we upsert flags instead of replacing them
-                        self.setCachedFeatureFlags(newFeatureFlags)
-                        self.setCachedFeatureFlagPayload(newFeatureFlagsPayloads)
+                        newFeatureFlags = cachedFeatureFlags.merging(featureFlags) { _, new in new }
+                        newFeatureFlagPayloads = cachedFeatureFlagsPayloads.merging(featureFlagPayloads) { _, new in new }
                     } else {
-                        self.setCachedFeatureFlags(featureFlags)
-                        self.setCachedFeatureFlagPayload(featureFlagPayloads)
+                        newFeatureFlags = featureFlags
+                        newFeatureFlagPayloads = featureFlagPayloads
                     }
-                }
 
-                self.notifyAndRelease()
+                    self.setCachedFeatureFlags(newFeatureFlags)
+                    self.setCachedFeatureFlagPayload(newFeatureFlagPayloads)
+
+                    // calculate a set of keys that were updated, added or removed
+                    //
+                    // Note: payload changes are not currently tracked, as this update mechanism
+                    //       is primarily used for capturing `$feature_flag_called` events tied to flag values.
```

Comment:
The goal is to clear the dict with flags when reloading/resetting/closing the SDK, but here we are comparing whether the keys and values changed, which adds more complexity.
JS SDK does not do that either, so I don't think we should do that either, the simpler the better here.
I'd expect to see a read flag event if I reloaded my flags.